arroyo-worker/src/operators/sources/kafka/mod.rs

```
@@ -164,74 +164,111 @@ where
+                                    if self.checkpoint(c, ctx).await {
+                                        return SourceFinishType::Immediate;
+                                    }
+                                },
+                                Some(ControlMessage::Stop { mode }) => {
+                                    info!("Stopping kafka source: {:?}", mode);
 
-                            match mode {
-                                StopMode::Graceful => {
-                                    return SourceFinishType::Graceful;
+                                    match mode {
+                                        StopMode::Graceful => {
+                                            return SourceFinishType::Graceful;
+                                        }
+                                        StopMode::Immediate => {
+                                            return SourceFinishType::Immediate;
+                                        }
+                                    }
                                 }
-                                StopMode::Immediate => {
-                                    return SourceFinishType::Immediate;
+                                None => {
+
                                 }
                             }
-                        }
-                        None => {
 
                         }
                     }
-
                 }
             }
+            Err(err) => {
+                ctx.control_tx
+                    .send(ControlResp::Error {
+                        operator_id: ctx.task_info.operator_id.clone(),
+                        task_index: ctx.task_info.task_index.clone(),
+                        message: "Error while getting kafka consumer".to_string(),
+                        details: format!("{:?}", err),
+                    })
+                    .await
+                    .unwrap();
+                panic!("Error while getting kafka consumer")
```

this should also report the actually error (so add `err` to the panic)